sound/soc/sof/ipc4-topology.c

```diff
@@ -976,6 +976,57 @@ static int sof_ipc4_widget_assign_instance_id(struct snd_sof_dev *sdev,
 	return 0;
 }
 
+static void sof_update_hw_param(struct snd_pcm_hw_params *params,
+				snd_pcm_hw_param_t var, unsigned int val)
+{
+	if (hw_is_mask(var)) {
+		struct snd_mask *m = hw_param_mask(params, var);
+
+		snd_mask_none(m);
+		snd_mask_set_format(m, val);
+	} else if (hw_is_interval(var)) {
+		struct snd_interval *i = hw_param_interval(params, var);
+
+		i->min = val;
+		i->max = val;
+		i->openmin = 0;
+		i->openmax = 0;
+		i->integer = 1;
```

Comment:
what do the last 3 lines above do and what are they needed for?

Reply:
@libinyang this sort of code doesn't seem be used in most drivers, i.e. there are exactly two occurrences of such code in the entire sound/soc tree. I would assert that this is needlessly complicated?

````
sound/soc$ git grep openmax
atmel/atmel_ssc_dai.c:          t.openmin = t.openmax = 0;
sti/sti_uniperif.c:     t.openmax = 0;
````

Reply:
@plbossart The ```integer``` in atmel and sti belongs the local variable ```struct snd_interval t```. The snd_interval_refine() will be called immediately after this setting. And the set of ```t.integer = 0``` is just letting the ```integer``` in the local ```struct snd_interval t``` not impact ```integer``` in the hw_param's ```snd_interval i```. 
However I think ```integer``` may be not necessary to set in our case as the ```integer``` should already set before and don't need to be changed.

Reply:
My point @libinyang is that virtually no one else has played with this sort of constraints. That makes me question if this is a wise direction. This is a very very obscure part of ALSA, do we even want to look into this?

Reply:
@plbossart Yes. I agree we do not touch ```integer```. Setting ```min```, ```max```, ```openmin``` and ```openmax``` is enough.

Reply:
you're not getting the point @libinyang. Even openmin and openmax are not used very often, and not at all outside of codecs, so it's unclear to me why the SOF driver would need to be the first one to use these fields.

Reply:
> you're not getting the point @libinyang. Even openmin and openmax are not used very often, and not at all outside of codecs, so it's unclear to me why the SOF driver would need to be the first one to use these fields.

I removed the openmin, openmax and integer updating in the new version as it is supposed that such setting should already be set correctly when open.